AWGL/GermlineEnrichment_TSC
# Publish v1.0.0

## Change description

### Change description
Currently multiple versions of this pipeline running on different panels, due to small differences in reporting of each panel:
- Familial Hypercholesterolemia (FH) – v2.5.6
- TruSightCancer (TSC) – v2.5.3
- TruSightOne (TSO) – was run on v2.5.3 but now run on DragenGE

There are also a few changes on the cluster versions of these pipelines that haven’t been committed to GitHub yet

This update split the TSC section of the GermlineEnrichment repository into its own repository.
At the same time new BED files were added for TSC and both repos were checked to make sure that they are in sync with any untracked changes made directly to the cluster.

### Related issues
#1 

### Justification for minor update
Doesn’t affect sensitivity – Code is already in place and validated, it’s just splitting the two pipelines and verifying that everything is as it should be.

---
## Testing description

### Testing description
- Check for local changes - UNIX diff command checks for differences between files, the following pipelines were comapred with diff:
  - GE 2.5.3 cluster and GE 2.5.3 GitHub - differences were pushed to GE_TSC GitHub repo
  - GE 2.5.6 cluster and GE latest GitHub - differences were pushed  to GE GitHub repo
- Add BED files to GE_TSC GitHub – checked by Familial Cancer and bioinformatics
- Run through old runs to verify updated pipelines work as expected. Check variants called in gVCF and CNV files using diff command to make sure sensitivity wasn’t affected.
  - TSC – New GE_TSC pipeline
  - TSO – New GE_TSC pipeline
  - FH – original GE pipeline with updates

### Test data location
Test runs - cluster (gen-01):
- TSC verification run - `/data/results/ 191007_NB551415_0037_AHKN7WAFXY/IlluminaTruSightCancer_updated`
- TSO verification run - `/data/results/ 200110_D00501_0399_AH5YCFBCX3/IlluminaTruSightOne_test`
- FH verification run - `/data/results/ 200316_M00766_0302_000000000-CVB4V/AgilentOGTFH_v2.6.7`

Validation docs:
- Files related to this validation are saved in this repo in `validations/v1.0.0`
- Also saved in `L:\Bioinformatics\validations\GermlineEnrichment_TSC_v1.0.0`
- This includes excel files going through the diff command output, plus the raw diff output

### Code review
Previously completed in pull request #2 

---
## Verification results
### Check for local changes - TSC
These differences were detected and added to the new GitHub repo (see Github commits for details):
- `mqRankSum` variable being used instead of hardcoded value
- Loop added to R script to catch when a gaps file is empty
- Extra BED files

These changes have been made and the diff command was repeated. The only remaining differences are in the updated filepaths, which is to be expected. This is recorded in `validations/v1.0.0/comparison_diff_code.xlsx`.

---
### Update TSC BED files
All BED files were checked by Familial cancer, checked BED files are stored in `L:\NGS ANALYSIS\TruSightCancer\Validation\BED files`

The updated GitHub BED files were double checked by EW to make sure they match the checked BED files. These were MET, BAP1, BRIP1, CHEK2, RAD51C, RAD51D

---
### Check for local changes – FH
There were two differences: 
- a commit to GitHub which isn’t on the cluster to prevent polygenic risk score being calculated on the NTC
- Panel config files for OGT on the cluster that aren’t present on GitHub.

These changes have been added to the original GermlineEnrichment repo and a new version (v2.5.7) has been created and added to the cluster.

---
### TSC verification run
GermlineEnrichment_TSC v1.0.0 has been tested with the TSC run `191007_NB551415_0037_AHKN7WAFXY`

Differences between the new and old runs were checked using the diff command and are shown in `validations/v1.0.0/TSC & TSO\comparison_diff_vcf.xlsx` and `validations/v1.0.0/TSC & TSO\comparison_diff_cnv.xlsx`.

Any differences did not affect pipeline sensitivity - different filepaths, different timestamps, or different regions due to updated BED files.

**NOTE:** A bug was identified in the CNV report section of the code, where a CNV covering multiple genes of interest was only getting labelled with one (see #3 ). This is unlikely to have caused an issue in the past as the ROI has only recently been expanded. This only affects TSC. This has been raised and is a fix is planned for the next release of the GE_TSC pipeline.

---
### TSO verification run
The same differences between the local GE v2.5.3 and the GitHub version for TSC were also present in TSO. 

GermlineEnrichment_TSC v1.0.0 has been tested with the TSO run `200110_D00501_0399_AH5YCFBCX3`

All gVCFs produced identical output except from the header line, which had different timestamps and filepaths. Output is shown in `validations/v1.0.0/TSC & TSO`

TSO is now run on DragenGE, but if the Dragen was to go down, TSO can be run through GermlineEnrichment_TSC v1.0.0 and give the same results as the previous version of GE that TSO was run on (v2.5.3).

---
### FH verification run

GermlineEnrichment v2.5.7 has been tested on FH run `200316_M00766_0302_000000000-CVB4V`

Changes to the GE panel that FH goes through were minimal.
CNV output files and gVCFs were compared, files are identical to the previous version (except for filepaths and time stamps in VCF header, which is to be expected). Output is shown in `validations/v1.0.0/FH`.
